staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -390,7 +397,7 @@ func (cnc *CloudNodeController) initializeNode(ctx context.Context, node *v1.Nod
 
 		// After adding, call UpdateNodeAddress to set the CloudProvider provided IPAddresses
 		// So that users do not see any significant delay in IP addresses being filled into the node
-		cnc.updateNodeAddress(ctx, curNode, instances)
+		cnc.updateNodeAddress(ctx, curNode)
```

Comment:
Can we actually go one step further. With the current change, you still be calling GetInstanceMetadata twice:
- in getNodeModifiersFromCloudProvider
- here

Can we actually call GetInstanceMetadata in this function and pass it to those two functions?

Reply:
Yes, I have thought it. But I plan to do it as a follow-up work, since this is large enough and tests need to be addressed.

Reply:
OK, cool. Can you please add a TODO so that we won't forget about it?

Reply:
Sure.

Reply:
Added.